libp2p/switch.nim

```diff
@@ -206,19 +206,19 @@ proc identify(s: Switch, conn: Connection) {.async, gcsafe.} =
 
     trace "identify: identified remote peer", peer = $conn.peerInfo
 
-proc mux(s: Switch, conn: Connection) {.async, gcsafe.} =
+proc mux(s: Switch, conn: Connection): Future[bool] {.async, gcsafe.} =
```

Comment:
does anything in the mux flow raise? if so, might as well use raises throughout - there are few things harder to understand than exceptions, but exceptions and error returns is one of them

Reply:
Actually you are right, there is a raise, I will stick to raises. 
The only issue is that the caller might want to close the connection and that needs signaling... handle and re-rise seems overkill.

Reply:
On the other hand the current raise will not clear the connection indeed.. for example......